src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -413,21 +419,38 @@ export class StudyViewPageStore
     }
     @action
     toggleNAValue = (uniqueKey: string): void => {
-        const filterSet = this.getDataBinFilterSet(uniqueKey);
-        const newFilter = _.clone(filterSet.get(uniqueKey)!);
-        newFilter.showNA = !this.isShowNAChecked(uniqueKey);
-        filterSet.set(uniqueKey, newFilter as any);
+        if (this._customDataBinFilterSet.has(uniqueKey)) {
+            const filterSet = this._customDataBinFilterSet.get(uniqueKey);
+            const newFilter = _.clone(filterSet!);
+            newFilter.showNA = !this.isShowNAChecked(uniqueKey);
+            this._customDataBinFilterSet.set(uniqueKey, newFilter as any);
+        } else {
+            const filterSet = this.getDataBinFilterSet(uniqueKey);
+            const newFilter = _.clone(filterSet.get(uniqueKey)!);
+            newFilter.showNA = !this.isShowNAChecked(uniqueKey);
+            filterSet.set(uniqueKey, newFilter as any);
+        }
     };
 
     public isShowNAChecked = (uniqueKey: string): boolean => {
-        const filter = this.getDataBinFilterSet(uniqueKey).get(uniqueKey)!;
-        const showNA = filter ? filter.showNA : undefined;
-
-        // Show NA bars by default
-        if (showNA === undefined) {
-            return true;
+        if (this._customDataBinFilterSet.has(uniqueKey)) {
+            const filter = this._customDataBinFilterSet.get(uniqueKey);
+            const showNA = filter ? filter.showNA : undefined;
```

Comment:
this logic seems kind of painful.  i would just do 
```
// default to true
return filter.showNA === undefined ? true : filter.showNA;
```

Reply:
Updated to : 
```
            const filter = this._customDataBinFilterSet.get(uniqueKey);
            return filter === undefined ? true : (filter!.showNA as boolean);
```